code/modules/clothing/gloves/miscellaneous.dm

```diff
@@ -315,3 +314,50 @@
 	. = ..()
 	if(mounted)
 		mounted.switch_firemodes()
+
+/obj/item/clothing/gloves/tesla
+	name = "tesla gauntlet"
+	desc = "A weaponized gauntlet capable of firing lightning bolts."
+	desc_fluff = "A tesla-based weapon created by the People's Republic of Adhomai as part of their Tesla Brigade program. Because of its long recharge time, the gauntlet is commonly \
+	used as an ancillary weapon."
+	icon_state = "tesla_glove_on"
+	item_state = "tesla_glove_on"
+	siemens_coefficient = 1
+	fingerprint_chance = 50
+	clipped = TRUE
+	species_restricted = list("exclude",BODYTYPE_GOLEM,BODYTYPE_VAURCA_BREEDER,BODYTYPE_VAURCA_WARFORM)
+	drop_sound = 'sound/items/drop/toolbox.ogg'
+	pickup_sound = 'sound/items/pickup/toolbox.ogg'
+	gender = NEUTER
+	var/charged = TRUE
+
+/obj/item/clothing/gloves/tesla/Touch(atom/A, mob/living/user, proximity)
+	if(!charged)
+		to_chat(user, SPAN_WARNING("\The [src] is still recharging."))
+		return
+
+	if(user.a_intent == I_HURT)
+		var/turf/T = get_turf(user)
+		user.visible_message(SPAN_DANGER("\The [user] crackles with energy!"))
+		var/obj/item/projectile/beam/tesla/LE = new (T)
+		LE.muzzle_type = /obj/effect/projectile/muzzle/tesla
+		LE.tracer_type = /obj/effect/projectile/tracer/tesla
+		LE.impact_type = /obj/effect/projectile/impact/tesla
```

Comment:
is it possible to turn this into a new projectile to not have to define this here